plugin/trino-memsql/src/test/java/io/trino/plugin/memsql/TestMemSqlTypeMapping.java

```diff
@@ -455,6 +461,8 @@ public void testMemSqlCreatedParameterizedVarcharUnicode()
                         createVarcharType(sampleUnicodeLiteral.length()), "CAST(" + sampleUnicodeLiteral + " AS varchar(" + sampleUnicodeLiteral.length() + "))")
                 .addRoundTrip("varchar(32) " + CHARACTER_SET_UTF8, sampleUnicodeLiteral, createVarcharType(32), "CAST(" + sampleUnicodeLiteral + " AS varchar(32))")
                 .addRoundTrip("varchar(20000) " + CHARACTER_SET_UTF8, sampleUnicodeLiteral, createVarcharType(20000), "CAST(" + sampleUnicodeLiteral + " AS varchar(20000))")
+                // MemSQL version >= 7.5 supports utf8mb4, but older versions store an empty character for a 4 bytes character
+                .addRoundTrip("varchar(1) " + CHARACTER_SET_UTF8, "'😂'", createVarcharType(1), "CAST('' AS varchar(1))")
```

Comment:
TheI would've expected the verifyPredicate to fail since Trino would search for the 4-byte string but none would be found? How does it pass?

Reply:
I suppose the above comment (L464) explains everything.